### PR TITLE
Library documentation improvements

### PR DESCRIPTION
**Pull request by mejroslav**
Once again I was working with `asab.library` and found some missing info in the documentation.

### CODE REVIEW

**Comment by eliska-n** [comment]
yes! I 100% approve it. A picture. With layers :heart: . But one unresolved question @mithunbharadwaj

**Review by eliska-n — APPROVED** [review]

File: asab/library/item.py [hunk 1]
```diff
@@ -3,30 +3,21 @@
...
+		disabled (bool): `True` if the Item is disabled, `False` otherwise. If the Item is disabled, `LibraryService.read(...)` will return `None`.
+		override (int): If `True`, this item is marked as an override for the providers with the same Item name.
+	"""
+
+	name: str
+	type: str
+	layer: int
+	providers: list
+	disabled: bool = False
+	override: int = 0  # Default value for override is False
```
[the remark this answers is not in the document]
Comment (mithunbharadwaj) [reply asab/library/item.py:23]: The 'override' parameter in the context of a library is an integer value, not a boolean. This parameter specifies the layer level at which the given provider operates.

File: docs/reference/library.md [hunk 2]
```diff
@@ -12,120 +12,163 @@ The library can also notify the ASAB microservice about changes, e.g. for automa
...
 The layers of the library are like slices of Swiss cheese layered on top of each other.
 Only if there is a hole in the top layer can you see the layer that shows through underneath.
 It means that files of the upper layer overwrite files with the same path in the lower layers.
 
-The first provider is responsible for providing '/.disabled.yaml' that controls the visibility of items. 
-If '/.disabled.yaml' is not present, then is considered empty.
-
-## Library service
+<figure markdown>
+  ![Illustration of Library layers](../images/library/asab-library.drawio.png)
```
Comment (ateska) [inline docs/reference/library.md:43]: 1) The name of the file should be "asab-library-layers.drawio.png"
2) Layers are numbered from 0, with "0" having a special meaning in asap-library micro service

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/library/item.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ateska) [reply asab/library/item.py:23]: Specify what is a value of "override" for "file1.txt" in the picture below - to add a clarity.

File: docs/reference/library.md [hunk 3]
```diff
@@ -12,120 +12,163 @@ The library can also notify the ASAB microservice about changes, e.g. for automa
...
-  + /folder2/folder2.3/
-    - /folder2/folder2.3/item4.json
-```
+	```
+	+ /folder1/
+		- /folder1/item1.yaml
+		- /folder1/item2.json
+	+ /folder2/
+		- /folder2/item3.yaml
+		+ /folder2/folder2.3/
```
Comment (ateska) [inline docs/reference/library.md:23]: Folders cannot have dots in their names in the library.

File: docs/reference/library.md [hunk 4]
```diff
@@ -12,120 +12,163 @@ The library can also notify the ASAB microservice about changes, e.g. for automa
...
 
-## Library path rules
+!!! warning "Library path rules"
 
-- Any path must start with "/", including the root path.
-- The folder path must end with "/".
-- The item path must end with a file extension (e.g. ".json").
+	- Any path must start with "/", including the root path.
+	- The folder path must end with "/".
+	- The item path must end with a file extension (e.g. ".json").
```
Comment (ateska) [inline docs/reference/library.md:31]: No dots in the folder names.

File: docs/reference/library.md [hunk 5]
```diff
@@ -12,120 +12,163 @@ The library can also notify the ASAB microservice about changes, e.g. for automa
...
-import asab.library
+	In the most applications, it is common to create the first layer with *Zookeeper* provider and the layers beneath with *git* or *libsreg* provider. This allows you to quickly modify items of the Library on the first layer.
 
-class MyApplication(asab.Application):
+## Disabling files
 
-    async def initialize(self):
-        self.LibraryService = asab.library.LibraryService(self, "LibraryService") #(1)!
-        self.PubSub.subscribe("Library.ready!", self.on_library_ready) #(2)!
+The library concept supports multi-tenancy. By default, all items of the library are visible for everyone, but you can disable some of them for specific tenants.
```
Comment (ateska) [inline docs/reference/library.md:53]: Disable & multi-tenancy are two topics (their interconnect here sure).

File: docs/reference/library.md [hunk 6]
```diff
@@ -12,120 +12,163 @@ The library can also notify the ASAB microservice about changes, e.g. for automa
...
-                    print("  - (DISABLED)")
+In the following example, `file1.txt` is disabled for **tenant-1** and **tenant-2**, `file2.txt` for **tenant-1** and `file3.txt` for every tenant.
 
-if __name__ == '__main__':
-    app = MyApplication()
-    app.run()
+```yaml title=".disabled.yaml"
+/file1.txt:
+	- tenant-1
+	- tenant-2
```
Comment (ateska) [inline docs/reference/library.md:62]: No TABs in YAML.

**Review by ateska — CHANGES_REQUESTED** [review]

**Pull request merged**